libs/api/domains/national-registry-x-road/src/lib/utils/computeCountryResidence.ts

```diff
@@ -0,0 +1,59 @@
+import { NationalRegistryResidence } from '../../models/nationalRegistryResidence.model'
+import { DAY } from './constants'
+
+/**
+ * Compute a summary of how many days you've spent as a resident of each country in
+ * the last 365 days
+ *
+ * @param history list from national id registry of residences
+ * @returns map of country code to days spent in country in the last 365 days
+ */
+export const computeCountryResidence = (
+  history: NationalRegistryResidence[],
+) => {
+  if (history.length < 1) {
+    return null
+  }
+
+  const simplified = history
+    .map(({ dateOfChange, country }) => ({
+      time: dateOfChange.getTime(),
+      country,
+    }))
+    .sort(({ time: a }, { time: b }) => {
+      // reversed order, make sure we get this right even if the national ID
+      // registry is out of order
+      return b - a
+    })
+
+  const currentDateTime = new Date()
+
+  const now = new Date(
+    currentDateTime.getFullYear(),
+    currentDateTime.getMonth(),
+    currentDateTime.getDate(),
+  ).getTime()
```

Comment:
A general question I am wondering about, is there any reason why you don't use days instead of ms?